nomad/state/state_store.go

```diff
@@ -4885,6 +4885,99 @@ func (s *StateStore) updateJobStabilityImpl(index uint64, namespace, jobID strin
 	return s.upsertJobImpl(index, nil, copy, true, txn)
 }
 
+func (s *StateStore) UpdateJobVersionTag(index uint64, namespace, jobID string, jobVersion uint64, name string, description string) error {
+	txn := s.db.WriteTxn(index)
+	defer txn.Abort()
+
+	if err := s.updateJobVersionTagImpl(index, namespace, jobID, jobVersion, name, description, txn); err != nil {
+		return err
+	}
+
+	return txn.Commit()
+}
+
+func (s *StateStore) updateJobVersionTagImpl(index uint64, namespace, jobID string, jobVersion uint64, name string, description string, txn *txn) error {
+	ws := memdb.NewWatchSet()
+
+	tag := &structs.JobTaggedVersion{
+		Name:        name,
+		Description: description,
+		TaggedTime:  time.Now().UnixNano(),
```

Comment:
You can't call `time.Now()` when writing in the state store, because the FSM Apply has to be deterministic across the servers. If you want the time, it has to be in the request args.